File: Legacy/SSSExampleVBnet/SSSExampleVBnet/ClickWindow/Form1.vb
Imports ScreenScraper

Public Class Form1
    Private m_tCapture As TextCaptureX
    Private m_tSelection As TSelection
    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        comboClick.SelectedIndex = 0
        comboMouseBtn.SelectedIndex = 0

        Try
            m_tCapture = ComFactory.Instance.NewTextCaptureX
            m_tSelection = ComFactory.Instance.NewTSelection
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim tSelInfo As TSelectionInfo
        Try
            tSelInfo = m_tSelection.Start(TS_SELECTION.tsSelectionWindow, TS_SELECTION_OPTIONS.tsSelFlagDefaultText)

        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try
        Dim strID As String
        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem
            uiElement.hwnd = tSelInfo.WindowHandle
            strID = uiElement.GetID(True)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
            Return
        End Try
        txtHandle.Text = uiElement.hwnd.ToString("X")
        txtID.Text = strID

    End Sub

    Private Sub button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button2.Click
        Dim uiElement As UIElem
        Try
            uiElement = ComFactory.Instance.NewUIElem
            uiElement.UseClientCoordinates = True
            uiElement.InitializeFromID(txtID.Text, True)

            Dim left, top, right, bottom As Integer
            uiElement.GetRectangle(left, top, right, bottom)

            Dim flags As Integer
            flags = GetClickFlags()
            uiElement.Click(left, top, flags)
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub
    Private Function GetClickFlags() As Integer
        Dim nRes As Integer
        nRes = UIE_ClickFlags.UIE_CF_MOVE_CURSOR

        If (comboMouseBtn.SelectedIndex = 0) Then
            nRes = nRes Or UIE_ClickFlags.UIE_CF_LEFT
        ElseIf (comboMouseBtn.SelectedIndex = 1) Then
            nRes = nRes Or UIE_ClickFlags.UIE_CF_MIDDLE
        ElseIf (comboMouseBtn.SelectedIndex = 2) Then
            nRes = nRes Or UIE_ClickFlags.UIE_CF_RIGHT
        End If

        If (comboClick.SelectedIndex = 0) Then
            nRes = nRes Or UIE_ClickFlags.UIE_CF_SINGLE
        ElseIf (comboClick.SelectedIndex = 1) Then
            nRes = nRes Or UIE_ClickFlags.UIE_CF_DOUBLE
        End If

        GetClickFlags = nRes
    End Function
End Class

File: _site/Samples/Legacy/SSSExampleCS/SSSExampleCS/ClickWindow/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

using ScreenScraper;

namespace ClickWindow
{
    public partial class Form1 : Form
    {
        TextCaptureX m_tCapture;
        TSelection m_tSelection;

        public Form1()
        {
            InitializeComponent();

            comboClick.SelectedIndex = 0;
            comboMouseBtn.SelectedIndex = 0;

            try
            {
                m_tCapture = ComFactory.Instance.NewTextCaptureX();
                m_tSelection = ComFactory.Instance.NewTSelection();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            TSelectionInfo tSelInfo = null;

            this.WindowState = FormWindowState.Minimized;
            try
            {
                tSelInfo = (TSelectionInfo)m_tSelection.Start((int)TS_SELECTION.tsSelectionUIElement, (int)TS_SELECTION_OPTIONS.tsSelFlagDefaultText);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
                return;
            }
            this.WindowState = FormWindowState.Normal;
            string strID = null;
            strID = tSelInfo.UIElementID;

            txtHandle.Text = tSelInfo.WindowHandle.ToString("X");
            txtID.Text = strID;
        }

        private int GetClickFlags()
        {
            int nRes = (int)UIE_ClickFlags.UIE_CF_MOVE_CURSOR;

            // get mouse button
            if (comboMouseBtn.SelectedIndex == 0)
            {
                nRes |= (int)UIE_ClickFlags.UIE_CF_LEFT;
            }
            else if (comboMouseBtn.SelectedIndex == 1)
            {
                nRes |= (int)UIE_ClickFlags.UIE_CF_MIDDLE;
            }
            else if (comboMouseBtn.SelectedIndex == 2)
            {
                nRes |= (int)UIE_ClickFlags.UIE_CF_RIGHT;
            }

            // get click type
            if (comboClick.SelectedIndex == 0)
            {
                nRes |= (int)UIE_ClickFlags.UIE_CF_SINGLE;
            }
            else if (comboClick.SelectedIndex == 1)
            {
                nRes |= (int)UIE_ClickFlags.UIE_CF_DOUBLE;
            }
            return nRes;
        }

        private void button2_Click(object sender, EventArgs e)
        {
            UIElem uiElem = null;
            try
            {
                uiElem = ComFactory.Instance.NewUIElem();
                uiElem.UseClientCoordinates = true;
                uiElem.InitializeFromID(txtID.Text, true);

                int left, top, right, bottom;
                uiElem.GetRectangle(out left, out top, out right, out bottom);

                int flags = GetClickFlags();
                uiElem.Click(2, 2, flags);
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }
    }
}
